template/template.go

```diff
@@ -76,3 +84,137 @@ func (t *Template) ExecuteHTMLString(html string, data interface{}) (string, err
 	err = tmpl.Execute(&buf, data)
 	return buf.String(), err
 }
+
+type FuncMap map[string]interface{}
+
+var DefaultFuncs = FuncMap{
+	"upper": strings.ToUpper,
+	"lower": strings.ToLower,
+	"title": strings.Title,
+	// pairs allows for in-order iteration of key/value pairs.
+	"pairs": func(m map[string]string) []Pair {
+		var (
+			pairs     = make([]Pair, 0, len(m))
+			keys      = make([]string, 0, len(m))
+			sortStart = 0
+		)
+		for k := range m {
+			if k == string(model.AlertNameLabel) {
+				keys = append([]string{k}, keys...)
+				sortStart = 1
+			} else {
+				keys = append(keys, k)
+			}
+		}
+		sort.Strings(keys[sortStart:])
+
+		for _, k := range keys {
+			pairs = append(pairs, Pair{k, m[k]})
+		}
+		return pairs
+	},
+	"firing": func(alerts []Alert) []Alert {
```

Comment:
For PagerDuty and friends, resolving alerts as they stop firing is completely reasonable (only happens if the whole group is resolved). For more info-level stuff like email, I agree that it's noise but not everyone will feel the same.

Generally, this way ensures that my last notification for a group always gives me the last updates. Consider a dashboard behind a webhook, which might want that information in a timely manner. An alternative option would be to always keep resolved and firing separate and send two notifications if there is a mix within a group. That's generally more noisy though and can also be misleading.

I'm open to alternatives. Any suggestions?